selene-lib/src/rules/high_cyclomatic_complexity.rs

```diff
@@ -0,0 +1,423 @@
+            }
+
+            complexity = count_expression_complexity(lhs, complexity);
+            complexity = count_expression_complexity(rhs, complexity);
+
+            complexity
+        }
+
+        ast::Expression::Parentheses { expression, .. } => {
+            count_expression_complexity(expression, complexity)
+        }
+
+        ast::Expression::UnaryOperator { expression, .. } => {
+            count_expression_complexity(expression, complexity)
+        }
+
+        #[cfg_attr(
+            feature = "force_exhaustive_checks",
+            deny(non_exhaustive_omitted_patterns)
+        )]
+        ast::Expression::Value { value, .. } => match &**value {
+            // visit_value already tracks this
+            ast::Value::Function(_) => complexity,
+
+            ast::Value::FunctionCall(call) => {
+                for suffix in call.suffixes() {
+                    complexity = count_suffix_complexity(suffix, complexity)
+                }
+
+                complexity
+            }
+
+            ast::Value::ParenthesesExpression(paren_expression) => {
+                count_expression_complexity(paren_expression, complexity)
+            }
+
+            ast::Value::Number(_) => complexity,
+            ast::Value::String(_) => complexity,
+            ast::Value::Symbol(_) => complexity,
+
+            ast::Value::TableConstructor(table) => count_table_complexity(table, complexity),
+
+            ast::Value::Var(ast::Var::Expression(var_expression)) => {
+                for suffix in var_expression.suffixes() {
+                    complexity = count_suffix_complexity(suffix, complexity)
+                }
+                complexity
+            }
+
+            ast::Value::Var(ast::Var::Name(_)) => complexity,
+
+            #[cfg(feature = "roblox")]
+            ast::Value::IfExpression(if_expression) => {
+                complexity += 1;
+                if let Some(else_if_expressions) = if_expression.else_if_expressions() {
+                    for else_if_expression in else_if_expressions {
+                        complexity += 1;
+                        complexity = count_expression_complexity(
+                            else_if_expression.expression(),
+                            complexity,
+                        );
+                    }
+                }
+                complexity
+            }
+
+            _ => complexity,
+        },
+
+        _ => complexity,
+    }
+}
+
+fn count_block_complexity(block: &ast::Block, starting_complexity: u16) -> u16 {
+    let mut complexity = starting_complexity;
+
+    // we don't immediately return from the matched blocks so that we can add in any complexity from the last statement
+    for statement in block.stmts() {
+        #[cfg_attr(
+            feature = "force_exhaustive_checks",
+            deny(non_exhaustive_omitted_patterns)
+        )]
+        match statement {
+            ast::Stmt::Assignment(assignment) => {
```

Comment:
Note that this doesn't consider cases Assignment's such as
`var[if x then y else z] = true`
You may need to look through the vars here